testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/adduser/AddUserTest.java

```diff
@@ -0,0 +1,160 @@
+        String authServerQualifier = suiteContext.getAuthServerInfo().getQualifier();
+        assertThat("Qualifier of AuthServer is empty or NULL !!", authServerQualifier, not(isEmptyOrNullString()));
+        assertThat("Controller isn't running.", controller.isStarted(authServerQualifier), is(true));
+
+        AddUser.main(new String[]{"-u", username, "-p", "password", "--sc", configDir});
+
+        //Read keycloak-add-user.json
+        List<RealmRepresentation> realms = JsonSerialization.readValue(new FileInputStream(new File(configDir, "keycloak-add-user.json")),
+                new TypeReference<List<RealmRepresentation>>() {
+                });
+
+        assertThat("File 'keycloak-add-user.json' is empty.",realms.size()>0,is(true));
+
+        //-----------------Get-Indexes-------------------//
+        int realmIndex=getRealmIndex(realmName,realms);
+        assertThat("Realm "+realmName+" not found.",realmIndex,is(not(-1)));
+
+        int userIndex=getUserIndex(username,realms.get(realmIndex).getUsers());
+        assertThat("User "+username+" not found",userIndex,is(not(-1)));
+
+
+        UserRepresentation user = realms.get(realmIndex).getUsers().get(userIndex);
+        assertThat("Username from Json file is wrong.", user.getUsername(), is(username));
+
+        //------------------Credentials-----------------------------//
+        assertThat("User Credentials are NULL", user.getCredentials().get(0), notNullValue());
+        CredentialRepresentation credentials = user.getCredentials().get(0);
+        assertThat("User Credentials have wrong Algorithm.", credentials.getAlgorithm(), is(Pbkdf2Sha256PasswordHashProviderFactory.ID));
+        assertThat("User Credentials have wrong Hash Iterations", credentials.getHashIterations(), is(100000));
+
+        //------------------Restart--Container---------------------//
+        controller.stop(authServerQualifier);
+        controller.start(authServerQualifier);
+
+        RealmResource realmResource = getAdminClient().realm(realmName);
+        user = realmResource.users().search(username).get(0);
+        assertThat("Username is wrong.", user.getUsername(), is(username));
+        UserResource userResource = realmResource.users().get(user.getId());
+
+        //--------------Roles-----------------------//
+        List<RoleRepresentation> realmRoles = userResource.roles().realmLevel().listAll();
+
+        assertRoles(realmRoles, "admin", "offline_access", Constants.AUTHZ_UMA_AUTHORIZATION);
+
+        List<ClientRepresentation> clients = realmResource.clients().findAll();
+        String accountId = null;
+        for (ClientRepresentation c : clients) {
+            if (c.getClientId().equals("account")) {
+                accountId = c.getId();
+            }
+        }
+
+        List<RoleRepresentation> accountRoles = userResource.roles().clientLevel(accountId).listAll();
+        assertRoles(accountRoles, "view-profile", "manage-account");
+
+        userResource.remove();
```

Comment:
if there is a need to remove the user it will be better to do it even in case the test fails. You can achieve that by adding 
```
try {
    //test code between lines 98-112
} finally {
    userResource.remove();
}
```

Reply:
It makes sense. Thanks.